include/aikido/perception/PerceptionModule.hpp

```diff
@@ -20,7 +20,7 @@ class PerceptionModule{
     /// \param[in] timeout The duration up to which to wait for the transform. Returns false if none of the markers get correctly transformed
     /// \param[in] timestamp Only detections more recent than this timestamp will be accepted. A timestamp of 0 greedily takes the first available message, and is the default behaviour. 
     /// \return bool Returns \c false if no detection observed, or if none of the detections has a more recent timestamp than the parameter. Returns \c true otherwise.
-    virtual bool detectObjects(std::vector<dart::dynamics::SkeletonPtr>& skeleton_list, ros::Duration timeout, ros::Time timestamp) = 0;
+    virtual bool detectObjects(dart::simulation::WorldPtr env, ros::Duration timeout, ros::Time timestamp) = 0;
```

Comment:
This might not be directly related to this PR, but it seems we have two styles for taking `std::shared_ptr`: (1) const reference and (2) copy and move (don't know if there is a better expression for this). Style (2) is used in some constructors and creators. (e.g., [this](https://github.com/personalrobotics/aikido/blob/master/src/distance/Weighted.cpp#L8-L10) and [this](https://github.com/personalrobotics/aikido/blob/master/include/aikido/distance/detail/defaults-impl.hpp#L27-L29)) In Style (2), we pass `std::shared_ptr` by value and move it to the destination.

For those cases (copying `std::shared_ptr` to the destination), it would be the same at the point of performance view ((1) copies it when assigning to the destination, (2) copies it when passing to the argument). But for the above function, (1) is the better way since we can avoid an unnecessary copy. So I would like to suggest consistently using Style (1).

Also, we could consider [providing an additional function that takes rvalue reference if necessary](http://stackoverflow.com/a/20909578).